### ISSUE TITLE
[core] : In Event CT, moving endDate field before startDate field breaks edit modal

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#17497]
Reported via Support: [99447](https://dotcms.zendesk.com/agent/tickets/99447)
Reproducible in dotCMS 5.1.6, and Demo Site (5.2.0)

By default, the **Event** Content Type provides fields for the Start and End dates for a given Event. However, if a Content Author decides to move the **End Date** field **BEFORE** the **Start Date** field, the Edit Modal will not display correctly and an error will be logged.

### Steps to reproduce the behavior:
1. Go to **Types & Tags > Content Types > Click the "Event" Content Type**
2. Move the the **End Date** field **BEFORE** the **Start Date** field.
3. Go to **Content > Calendar > Add Event**.
4. The following error is displayed:
```
An unexpected system error occurred.

An exception occurred processing JSP page [/html/portal/view_portlet_inc.jsp] at line [87] 84: 85: boolean portletException = false; 86: 87: cachePortlet.render(renderRequest, renderResponse); 88: 89: SessionMessages.clear(renderRequest); 90: SessionErrors.clear(renderRequest); Stacktrace:
```

### Expected behavior
The order of the fields should NOT affect the inner workings of the **Calendar** portlet.

### Desktop:
 - OS: Not relevant.
 - Browser  Not relevant.
 - Version Not relevant.

## Acceptance Criteria

1. Unit test
2. Work in all the supported browsers (don't forget IE11 and iPad)

**Comment by wezell** [comment]
Just bit a customer on ticket id: 100363

**Comment by nollymar** [comment]
Fixed. PR: https://github.com/dotCMS/core/pull/17947

**Comment by dsilvam** [comment]
*Passed Internal QA*: Add Event functionality working without issues after moving the the End Date field BEFORE the Start Date field on the _Event_ content type.

**Comment by bryanboza** [comment]
Fixed, unable to reproduce on release-5.2.6 // Postgres // FF. Related card reported here #18008
